### PR TITLE
TAXII2 Logs

### PR DESCRIPTION
**Pull request by tkatzir**
## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://jira-dc.paloaltonetworks.com/browse/XSUP-33634

### CODE REVIEW

File: Packs/ApiModules/Scripts/TAXII2ApiModule/TAXII2ApiModule.py [hunk 1]
```diff
@@ -30,7 +31,11 @@ def filter(self, record):
         return False
 
 
+# Make sure we have only one SuppressWarningFilter
 v21_logger = logging.getLogger("taxii2client.v21")
+for current_filter in list(v21_logger.filters):    # pragma: no cover
+    if 'SuppressWarningFilter' in type(current_filter).__name__:
+        v21_logger.removeFilter(current_filter)
```
Comment (AradCarmi) [inline Packs/ApiModules/Scripts/TAXII2ApiModule/TAXII2ApiModule.py:38]: Won't that remove all the filters?

File: Packs/ApiModules/Scripts/TAXII2ApiModule/TAXII2ApiModule.py [hunk 2]
```diff
@@ -382,7 +387,7 @@ def init_collection_to_fetch(self, collection_to_fetch=None):
                     break
             if not collection_found:
                 raise DemistoException(
-                    "Could not find the provided Collection name in the available collections. "
+                    "Could not find the provided Collection name ({collection_to_fetch}) in the available collections. "
```
Comment (AradCarmi) [inline Packs/ApiModules/Scripts/TAXII2ApiModule/TAXII2ApiModule.py:390]:
```suggestion
                    f"Could not find the provided Collection name {collection_to_fetch} in the available collections. "
```

File: Packs/ApiModules/Scripts/TAXII2ApiModule/TAXII2ApiModule.py [hunk 3]
```diff
@@ -1219,20 +1224,35 @@ def load_stix_objects_from_envelope(self, envelopes: types.GeneratorType, limit:
...
         indicators = []
         relationships_lst = []
+        parsed_objects_counter: dict[str, int] = {}
         try:
             for envelope in envelopes:
-                stix_objects = envelope.get("objects")
-                if not stix_objects:
-                    # no fetched objects
-                    break
+                # demisto.debug(f'parse_generator_type_envelope, trying to parse the envelope: {envelope}')
```
Comment (AradCarmi) [inline Packs/ApiModules/Scripts/TAXII2ApiModule/TAXII2ApiModule.py:1244]:
```suggestion
```

File: Packs/ApiModules/Scripts/TAXII2ApiModule/TAXII2ApiModule.py [hunk 4]
```diff
@@ -1219,20 +1224,35 @@ def load_stix_objects_from_envelope(self, envelopes: types.GeneratorType, limit:
...
+    def increase_count(self, counter: dict[str, int], id: str):
+        if id in counter:
+            counter[id] = counter[id] + 1
+        else:
+            counter[id] = 1
+
     def parse_generator_type_envelope(self, envelopes: types.GeneratorType, parse_objects_func, limit: int = -1):
         indicators = []
         relationships_lst = []
+        parsed_objects_counter: dict[str, int] = {}
```
Comment (AradCarmi) [inline Packs/ApiModules/Scripts/TAXII2ApiModule/TAXII2ApiModule.py:1241]:
```suggestion
        parsed_objects_counter: Dict[str, int] = {}
```

File: Packs/ApiModules/Scripts/TAXII2ApiModule/TAXII2ApiModule.py [hunk 5]
```diff
@@ -1219,20 +1224,35 @@ def load_stix_objects_from_envelope(self, envelopes: types.GeneratorType, limit:
         if relationships_lst:
             indicators.extend(self.parse_relationships(relationships_lst))
         demisto.debug(
-            f"TAXII 2 Feed has extracted {len(indicators)} indicators"
+            f"TAXII 2 Feed has extracted {len(list(indicators))} indicators"
         )
 
         return indicators
 
+    def increase_count(self, counter: dict[str, int], id: str):
```
Comment (AradCarmi) [inline Packs/ApiModules/Scripts/TAXII2ApiModule/TAXII2ApiModule.py:1232]:
```suggestion
    def increase_count(self, counter: Dict[str, int], id: str):
```

**Review by AradCarmi — COMMENTED** [review]
Nice!

File: Packs/ApiModules/Scripts/TAXII2ApiModule/TAXII2ApiModule.py (hunk 1 above)
Comment (tkatzir) [reply Packs/ApiModules/Scripts/TAXII2ApiModule/TAXII2ApiModule.py:38]: I hope we're the only ones with the `SuppressWarningFilter ` filter class.

File: Packs/ApiModules/Scripts/TAXII2ApiModule/TAXII2ApiModule.py [hunk 6]
```diff
@@ -1219,20 +1226,34 @@ def load_stix_objects_from_envelope(self, envelopes: types.GeneratorType, limit:
...
+    def increase_count(self, counter: Dict[str, int], id: str):
+        if id in counter:
+            counter[id] = counter[id] + 1
+        else:
+            counter[id] = 1
+
     def parse_generator_type_envelope(self, envelopes: types.GeneratorType, parse_objects_func, limit: int = -1):
         indicators = []
         relationships_lst = []
+        parsed_objects_counter: Dict[str, int] = {}
```
Comment (DeanArbel) [inline Packs/ApiModules/Scripts/TAXII2ApiModule/TAXII2ApiModule.py:1243]: Please add a comment about this being used for logging purposes

File: Packs/ApiModules/Scripts/TAXII2ApiModule/TAXII2ApiModule.py [hunk 7]
```diff
@@ -30,8 +31,14 @@ def filter(self, record):
...
 
+# Make sure we have only one XsoarSuppressWarningFilter
 v21_logger = logging.getLogger("taxii2client.v21")
-v21_logger.addFilter(SuppressWarningFilter())
+demisto.debug(f'Logging Filters before cleaning: {v21_logger.filters=}')
+for current_filter in list(v21_logger.filters):    # pragma: no cover
+    if 'XsoarSuppressWarningFilter' in type(current_filter).__name__:
+        v21_logger.removeFilter(current_filter)
+v21_logger.addFilter(XsoarSuppressWarningFilter())
+demisto.debug(f'Logging Filters: {v21_logger.filters=}')
```
Comment (DeanArbel) [inline Packs/ApiModules/Scripts/TAXII2ApiModule/TAXII2ApiModule.py:41]: Do you have to remove the filter if you add it later?
Maybe an if condition/function in that case?

**Review by DeanArbel — APPROVED** [review]
Looks good, just small notes

File: Packs/ApiModules/Scripts/TAXII2ApiModule/TAXII2ApiModule.py (hunk 7 above)
Comment (tkatzir) [reply Packs/ApiModules/Scripts/TAXII2ApiModule/TAXII2ApiModule.py:41]: The root cause is that there are "old" filters, from previous runs, with no real handle to them.
I guess it's the way the `looper` is implemented.

File: Packs/ApiModules/Scripts/TAXII2ApiModule/TAXII2ApiModule.py (hunk 6 above)
Comment (tkatzir) [reply Packs/ApiModules/Scripts/TAXII2ApiModule/TAXII2ApiModule.py:1244]: [Done](https://github.com/demisto/content/pull/33061/commits/39693e56b65bbdb500e99662e87d81026a11cf43).

**Comment by tkatzir** [comment]
Force merged due to unrelated build issues.

**Pull request merged**